doc/api/n-api.md

```diff
@@ -3835,6 +3881,201 @@ NAPI_EXTERN napi_status napi_get_uv_event_loop(napi_env env,
 - `[in] env`: The environment that the API is invoked under.
 - `[out] loop`: The current libuv loop instance.
 
+<!-- it's very convenient to have all the anchors indexed -->
+<!--lint disable no-unused-definitions remark-lint-->
+## Asynchronous Thread-safe Function Calls
+
+JavaScript functions can normally only be called from a native addon's main
+thread. If an addon creates additional threads, then N-API functions that
+require a `napi_env`, `napi_value`, or `napi_ref` must not be called from those
+threads.
+
+This API provides the type `napi_threadsafe_function` as well as APIs to create,
+destroy, and call objects of this type. `napi_create_threadsafe_function()`
+creates a permanent reference to a `napi_value` that holds a JavaScript
+function, and uses `uv_async_t`, `uv_mutex_t`, `u_cond_t`, and `uv_idle_t` from
```

Comment:
permanent?  Maybe persistent, as I assume it is not permanent in the sense of never going away.